csharp/autobuilder/Semmle.Autobuild.CSharp/CSharpAutobuilder.cs

```diff
@@ -25,7 +26,8 @@ public class CSharpAutobuildOptions : AutobuildOptionsShared
         public CSharpAutobuildOptions(IBuildActions actions) : base(actions)
         {
             Buildless = actions.GetEnvironmentVariable(lgtmPrefix + "BUILDLESS").AsBool("buildless", false) ||
-                actions.GetEnvironmentVariable(extractorOptionPrefix + "BUILDLESS").AsBool("buildless", false);
+                actions.GetEnvironmentVariable(extractorOptionPrefix + "BUILDLESS").AsBool("buildless", false) ||
+                actions.GetEnvironmentVariable(buildModeEnvironmentVariable) == "none";
```

Comment:
Do we have to cover the different upper/lower casing of `nOnE`?

Reply:
The CLI will always set this to lowercase, and Java only accepts lowercase `none`, but is there a convention in the C# autobuilder that casing of environment variable values doesn't matter?

Reply:
Also, I thought this method would return null if the environment variable was not defined since it returns a `string?`, but it seems to be [throwing an exception](https://github.com/github/codeql/actions/runs/7616201809/job/20742424169?pr=15402).  What would be the right thing to do here to get an environment variable that might not be set?

Reply:
Regarding casing: The above `.AsBool` calls do handle different cases. But I don't mind forcing the casing of `none`.

Regarding exceptions: I think your assumption is correct that if the env var doesn't exist, the method returns `null`. Except in the tests, where we explicitly check if the variable exists: https://github.com/github/codeql/blob/main/csharp/autobuilder/Semmle.Autobuild.CSharp.Tests/BuildScripts.cs#L117-L123. I'd change this method.